src/main/java/ladder/domain/GameResult.java

```diff
@@ -0,0 +1,72 @@
+package ladder.domain;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class GameResult {
+
+    private final Map<User, LadderResult> result;
+
+    private GameResult(Users users, Ladder ladder, LadderResults ladderResults) {
+        result = game(users, ladder, ladderResults);
+    }
+
+    private Map<User, LadderResult> game(Users users, Ladder ladder, LadderResults ladderResults) {
+        Map<User, LadderResult> result = new LinkedHashMap<>();
+        users.getAllWithLocation()
+                .forEach(userLocation -> gameForEachUser(ladder, ladderResults, result, userLocation));
+        return result;
+    }
+
+    private void gameForEachUser(Ladder ladder, LadderResults ladderResults,
+                                 Map<User, LadderResult> result, UserLocation userLocation) {
+        User user = userLocation.getUser();
+        Location location = userLocation.getLocation();
+        ladder.getLines().forEach(line -> moveToLeftOrRight(location, line));
+        result.put(user, ladderResults.get(location.now()));
+    }
+
+    private void moveToLeftOrRight(Location location, Line line) {
+        if (isMovable(location.atLeft(), line)) {
+            location.toLeft();
+            return;
+        }
+        if (isMovable(location.now(), line)) {
+            location.toRight();
+        }
+    }
+
+    private boolean isMovable(int pos, Line line) {
```

Comment:
```suggestion
    private boolean isMovable(int position, Line line) {
```

개인적으론 축약하지 않는 것을 좋아하는데요
축약을 하여 작성하였을 때의 어떠한 장단점이 있을지 고민해보아요 :)

Reply:
동의 합니다!
알고리즘 짤 때 쓰던 습관이 베어서 저도 모르게 저렇게 했는데 변경해둘게요!